presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PickLayout.java

```diff
@@ -70,7 +68,7 @@ protected PlanNode visitPlan(PlanNode node, RewriteContext<Void> context)
         @Override
         public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
         {
-            if (node.getSource() instanceof TableScanNode && !((TableScanNode) node.getSource()).getLayout().isPresent()) {
```

Comment:
If we push down predicate into table scan, do we retain it in full as a filter above it?
Otherwise we risk that we clobber a filter previously pushed-down (and otherwise eliminated from the plan) with a new filter. (This, of course, would require a more complex query plan, where filter can appear above table scan twice.)

Reply:
@findepi I don't follow. Can you give an example? what kind of (deterministic) filters can't be combined together?